src/govuk/i18n.mjs

```diff
@@ -15,19 +15,62 @@ function I18n (translations) {
  * returns the appropriate string.
  *
  * @param    {String}  lookupKey  - The lookup key of the string to use.
+ * @param    {Object}  options    - Any options passed with the translation string, e.g: for string interpolation.
  * @returns  {String}             - The appropriate translation string.
  */
-I18n.prototype.t = function (lookupKey) {
+I18n.prototype.t = function (lookupKey, options) {
   if (!lookupKey) {
     // Print a console error if no lookup key has been provided
     throw new Error('i18n lookup key missing.')
   } else if (lookupKey in this.translations) {
-    // Return the translation string if it exists in the object
-    return this.translations[lookupKey]
+    // Fetch the translation string for that lookup key
+    var translationString = this.translations[lookupKey]
+
+    // Check for ${} placeholders in the translation string
+    if (translationString.match(/\${(.\S+)}/)) {
```

Comment:
Something I've noticed is that the format for placeholders here is `${variableName}`, changed from `%{variableName}` in my previous spikes. 

I used `%{}` previously as this is the same format used in [govuk_publishing_components](https://github.com/alphagov/govuk_publishing_components/blob/main/config/locales/cy.yml#L31). 

My concern with using `${}` is it's the same format as JavaScript's own template literals, and that feels like it may create conflicts in the code implemented by services and potentially our own when components are refactored to take advantage of newer JavaScript features. 

For a bit of a contrived example:

```javascript
const prefix = 'You have'
new CharacterCount($element, {
  i18n: {
    charactersUnderLimitOne: `${prefix} ${count} character remaining`,
    charactersUnderLimitOther: `${prefix} ${count} characters remaining`,
    charactersOverLimitOne: `${prefix} ${count} character too many`,
    charactersOverLimitOther: `${prefix} ${count} characters too many`
  }
}).init();
```

In this instance, `${prefix}` is being used to interpolate a string into a JavaScript template literal, but `${count}` is expected to be replaced by our i18n function. Instead, this is likely to produce an error stating that the `count` variable doesn't exist in this context.

Reply:
That's a good point, I'll change it 👍